pkg/providers/content.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package providers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/coreos/coreos-cryptagent/pkg/config"
+	_ "github.com/coreos/ignition/config/types"
+)
+
+type content struct {
+	source                 string
+	responseHeadersTimeout int
+	totalTimeout           int
+	certAuths              []string
+}
+
+func contentFromConfigV1(cfg *config.ProviderJSON) (*content, error) {
+	if cfg == nil {
+		return nil, errors.New("nil config")
+	}
+	if cfg.Kind != config.ProviderContentV1 {
+		return nil, fmt.Errorf("expected kind %q, got %q", config.ProviderContentV1, cfg.Kind)
+	}
+
+	value, ok := cfg.Value.(config.ContentV1)
+	if !ok {
+		return nil, errors.New("not a ContentV1 value")
+	}
+
+	c := content{
+		source: value.Source,
+	}
+	if value.Timeouts != nil {
+		c.totalTimeout = value.Timeouts.HTTPTotal
+		c.responseHeadersTimeout = value.Timeouts.HTTPResponseHeaders
+	}
+	for _, ca := range value.CertificateAuthorities {
+		if ca.Authority != "" {
+			c.certAuths = append(c.certAuths, ca.Authority)
+		}
+	}
+
+	return &c, nil
+}
+
+/*
+func contentFromIgnitionV230(ks types.LuksKeyslot, ign types.Ignition) (*content, error) {
```

Comment:
This depends on new Ignition public types introduced here https://github.com/coreos/ignition/pull/515, once merged.